### PR TITLE
feat: add folder views

### PR DESCRIPTION
**Pull request by seaerchin**
### TL;DR

- Updated focus rings in `CmsSideNav` component for better accessibility.
- Fixed pagination display issues in `Datatable` component.
- Added new route for folder details page under `/sites/[siteId]/folders/[folderId]/`.
- Adjusted button size in site page to maintain consistency.
- Updated `folder.router.ts` to include title and permalink in the response.

### What changed?

- Changed `p` to `m` in `CmsSideNav` component to ensure proper margin for focus ring.
- Removed unnecessary check on `totalRowCountString` in `Datatable` component.
- Added components and hooks for the new folder details page.
- Modified button size in `/sites/[siteId]/` from `xs` to `md`.
- Updated folder router to return title and permalink.

### How to test?

1. **Focus Ring**: Navigate to the CMS sidebar and check focus ring behavior on navigation items.
2. **Pagination**: Verify pagination works correctly in the `Datatable` component without hidden elements.
3. **Folder Page**: Access a folder using the new route and ensure all elements load correctly.
4. **Button Size**: Check the button size consistency on the site page.
5. **Router**: Confirm the API returns title and permalink along with other details for a folder.

### Why make this change?

- Improve the accessibility and user experience of the navigation components.
- Ensure pagination elements are displayed correctly.
- Introduce the folder details page for enhanced content management.
- Maintain UI consistency across the application.
- Provide more data context in API responses for better frontend handling.

### CODE REVIEW

**Comment by seaerchin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/437?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#437** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/437?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#436** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/436?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#418** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/418?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#417** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/417?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#403** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/403?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @seaerchin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/studio/src/components/CmsSidebar/CmsSideNav.tsx [hunk 1]
```diff
@@ -120,11 +120,19 @@
...
+                    boxShadow: "none !important",
+                    outline: `2px solid var(--chakra-colors-utility-focus-default)`,
+                    outlineOffset: "0.125rem",
+                    _dark: {
+                      outline: `2px solid var(--chakra-colors-utility-focus-inverse)`,
+                    },
+                  }}
                   _hover={{
                     backgroundColor: "interaction.muted.main.hover",
                   }}
```
Comment (karrui) [inline apps/studio/src/components/CmsSidebar/CmsSideNav.tsx:138]: <i class='graphite__hidden'>[Re: line +145]<br/><br/></i>

lint error here
```
Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator
```<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/437?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

File: apps/studio/src/components/Datatable/Datatable.tsx [hunk 2]
```diff
@@ -136,18 +136,11 @@ export const Datatable = <T extends object>({
         </Table>
       </Box>
       <Flex py="1rem" gap="1rem">
-        {totalRowCountString && (
```
Comment (karrui) [inline apps/studio/src/components/Datatable/Datatable.tsx:139]: Should keep right

File: apps/studio/src/components/Datatable/Datatable.tsx [hunk 3]
```diff
@@ -136,18 +136,11 @@ export const Datatable = <T extends object>({
...
-          <Text textStyle="caption-2" color="base.content.medium">
-            {totalRowCountString}
-          </Text>
-        )}
         {pagination && (
```
Comment (karrui) [inline apps/studio/src/components/Datatable/Datatable.tsx:144]: this is the one that may need to change. Only render if there are rows:

```ts
totalRowCount > 0 && pagination && <...>
```

**Review by karrui — APPROVED** [review]

File: apps/studio/src/components/Datatable/Datatable.tsx (hunk 2 above)
Comment (seaerchin) [reply apps/studio/src/components/Datatable/Datatable.tsx:139]: it wasn't in the design - see [here](https://www.figma.com/design/g7LnLn4IUcUC9dYxgsHEJM/IsomerCMS-V2-(working-title%3A-%22Isomer-Studio%22)?node-id=7336-40934&t=k3MIjEPJKGyVheCA-4)

File: apps/studio/src/components/Datatable/Datatable.tsx (hunk 3 above)
Comment (karrui) [reply apps/studio/src/components/Datatable/Datatable.tsx:144]: @seaerchin this still requires change, since it may look weird if there are no rows

**Review by karrui — APPROVED** [review]

**Comment by seaerchin** [comment]
### Merge activity

* **Aug 12, 11:51 AM EDT**: @seaerchin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/437).

**Pull request merged**